flytekit/configuration/__init__.py

```diff
@@ -285,7 +285,7 @@ def validate_image(_: typing.Any, param: str, values: tuple) -> ImageConfig:
                 images.append(img)
 
         if default_image is None:
-            default_image_str = os.environ.get("FLYTE_INTERNAL_IMAGE", DefaultImages.default_image())
+            default_image_str = DefaultImages.default_image()
```

Comment:
This code block is entered when `default_image is None` however for e.g. `pyflyte register` the default_image will always be something like `cr.flyte.org/...` and so the `FLYTE_INTERNAL_IMAGE` env var never takes effect.

Reply:
@ddl-rliu `os.environ.get` returns `DefaultImages.default_image()` only if `FLYTE_INTERNAL_IMAGE` doesn't exist, correct?

Reply:
Right! In particular, note that the existence of `FLYTE_INTERNAL_IMAGE` is checked here https://github.com/flyteorg/flytekit/pull/2223/files#diff-e3e9429cda3e0f0d375729117b25595b280937f3a04e90f1efc7ee2c1a164008R38 and so it would not be necessary to doubly-check that env var here in `configuration/__init__.py`